src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -43,14 +44,85 @@ public partial class GenericVectorTests
         private void TestConstructorWithSpan<T>() where T : struct
         {
             T[] values = GenerateRandomValuesForVector<T>().ToArray();
-            var valueSpan = new Span<T>(values);
+            Vector<T> vector;
 
-            var vector = new Vector<T>(valueSpan);
+            // Span<T> ctor
+            vector = new Vector<T>(new Span<T>(values));
             ValidateVector(vector,
                 (index, val) =>
                 {
                     Assert.Equal(values[index], val);
                 });
+
+            // ReadOnlySpan<T> ctor
+            vector = new Vector<T>(new ReadOnlySpan<T>(values));
```

Comment:
nit: Might be nice to have these as separate unit tests, since are testing different inputs. Otherwise, one failure prevents the other checks from happening.